fe/fe-core/src/test/java/com/starrocks/analysis/RefreshMaterializedViewTest.java

```diff
@@ -1245,4 +1246,52 @@ public void testTruncateTableInDiffDb() throws Exception {
         count = mvEntity.histRefreshJobDuration.getCount();
         Assert.assertEquals(1, count);
     }
+
+    @Test
+    public void testDropPartitionTableInDiffDb() throws Exception {
+        starRocksAssert
+                .createDatabaseIfNotExists("drop_db")
+                .useDatabase("drop_db")
+                .withTable("CREATE TABLE tbl_with_mv\n" +
+                        "(\n" +
+                        "    k1 date,\n" +
+                        "    k2 int,\n" +
+                        "    v1 int sum\n" +
+                        ")\n" +
+                        "PARTITION BY RANGE(k1)\n" +
+                        "(\n" +
+                        "    PARTITION p1 values [('2022-02-01'),('2022-02-16')),\n" +
+                        "    PARTITION p2 values [('2022-02-16'),('2022-03-01'))\n" +
+                        ")\n" +
+                        "DISTRIBUTED BY HASH(k2) BUCKETS 3\n" +
+                        "PROPERTIES('replication_num' = '1');");
+
+        starRocksAssert.createDatabaseIfNotExists("drop_mv_db")
+                .useDatabase("drop_mv_db")
+                .withMaterializedView("CREATE MATERIALIZED VIEW test_mv\n"
+                        + "DISTRIBUTED BY HASH(`k2`)\n"
+                        + "REFRESH ASYNC\n"
+                        + "AS select k1, k2, v1  from drop_db.tbl_with_mv;");
+
+        executeInsertSql(connectContext, "insert into drop_db.tbl_with_mv partition(p2) values(\"2022-02-20\", 2, 10)");
+        MaterializedView mv1 = getMv("drop_mv_db", "test_mv");
+        MaterializedViewMetricsEntity mvEntity =
+                (MaterializedViewMetricsEntity) MaterializedViewMetricsRegistry.getInstance().getMetricsEntity(mv1.getMvId());
+        long count = mvEntity.histRefreshJobDuration.getCount();
+        Assert.assertEquals(0, count);
+
+        OlapTable table = (OlapTable) getTable("drop_db", "tbl_with_mv");
+        Partition p1 = table.getPartition("p1");
+        DropPartitionClause dropPartitionClause = new DropPartitionClause(false, p1.getName(), false, true);
+        dropPartitionClause.setResolvedPartitionNames(ImmutableList.of(p1.getName()));
+        Database db = GlobalStateMgr.getCurrentState().getLocalMetastore().getDb("drop_db");
+        GlobalStateMgr.getCurrentState().getLocalMetastore().dropPartition(db, table, dropPartitionClause);
+        // sleep 3s, wait for the refresh job to complete
+        Thread.sleep(3000);
```

Comment:
This may cause unstable tests.

You can use `StarRocksAssert.waitRefreshFinished` instead.

```
    /**
     * Wait the input mv refresh task finished.
     * @param mvId: mv id
     * @return true if the mv refresh task finished, otherwise false.
     */
    public boolean waitRefreshFinished(long mvId) {
        TaskManager tm = GlobalStateMgr.getCurrentState().getTaskManager();
        Task task = tm.getTask(TaskBuilder.getMvTaskName(mvId));
        Assert.assertTrue(task != null);
        TaskRunManager taskRunManager = tm.getTaskRunManager();
        TaskRunScheduler taskRunScheduler = taskRunManager.getTaskRunScheduler();
        TaskRun taskRun = taskRunScheduler.getRunnableTaskRun(task.getId());
        int maxTimes = 1200;
        int count = 0;
        while (taskRun != null && count < maxTimes) {
            ThreadUtil.sleepAtLeastIgnoreInterrupts(500L);
            taskRun = taskRunScheduler.getRunnableTaskRun(task.getId());
            count += 1;
        }
        return taskRun == null;
    }
```

Reply:
Thanks, I have modified it.